src/blockdata/script.rs

```diff
@@ -329,10 +329,21 @@ impl Script {
     /// opcodes, datapushes and errors. At most one error will be returned and then the
     /// iterator will end. To instead iterate over the script as sequence of bytes, treat
     /// it as a slice using `script[..]` or convert it to a vector using `into_bytes()`.
-    pub fn iter(&self, enforce_minimal: bool) -> Instructions {
+    ///
+    /// To force minimal pushes, use [instructions_minimal].
+    pub fn instructions(&self) -> Instructions {
         Instructions {
             data: &self.0[..],
-            enforce_minimal: enforce_minimal,
+            enforce_minimal: false,
+        }
+    }
+
+    /// Iterate over the script in the form of `Instruction`s while enforcing
+    /// minimal pushes.
+    pub fn instructions_minimal(&self) -> Instructions {
```

Comment:
Nit: What do you think about naming this `minimal_instructions` instead? The motivation would be to be closer to how you would use it in an English sentence :)

Reply:
The advantage of current version is autocomplete-friendiness.

Alternative would be to have `fn enforce_minimal(self) - >Self` on `Instructions` itself, so it would be `instructions().enforce_minimal()`. Less discoverable, maybe, but reads better, and generalizes better in case further options would be introduced in the future.

Reply:
@thomaseizinger Hmm, I think the `<method>_<specialization thereof>()` pattern is common, no? Like `iter_mut` instead of `mut_iter` (as per mutable iterator)..

Reply:
Fair points :)